### ISSUE TITLE
Add new action EC2 hibernate to custodian

### ISSUE THREAD

**Issue by kumarsiyer** [cloud-custodian/cloud-custodian#3326]
https://aws.amazon.com/blogs/aws/new-hibernate-your-ec2-instances/

**Comment by kumarsiyer** [comment]
Yes thanks @kapilt I have been also reading about pre-requisites. Will provide more details around the use case and it's benefits out of it

**Comment by amit-chhatbar** [comment]
We have a mainframe application which we migrated to cloud and its taking about 20-25 min to come up on restart. This will be an ideal case for to add this capability to Cloud Custodian.

It has all pre-req and ready to Hibernate. Let me know if there is any update in Custodian.

-Supported instance families - M5
-Instance size - not a bare metal instances.
-Instance RAM size - 64G
-Supported AMIs (must be an HVM AMI that supports hibernation) : Amazon Linux 2 AMI released 2019.08.29 or later.
-Encrypted EBS
etc..

**Comment by tjstansell** [comment]
This came up in discussions around here and it looks like the boto3 `stop_instances()` docs imply this is relatively safe to add.  It's just a flag:

> Hibernate (boolean) --
> Hibernates the instance if the instance was enabled for hibernation at launch. If the instance cannot hibernate successfully, a normal shutdown occurs. For more information, see Hibernate Your Instance in the Amazon Elastic Compute Cloud User Guide .
> 
> Default: false

So, cloud-custodian could support passing a flag to `stop` or simply add a `hibernate` action that calls `stop` with hibernation enabled.

**Comment by tjstansell** [comment]
This should be closed now that #5348 has been merged.  You would simply pass the `hibernate` flag to the `stop` action:
```
  actions:
    - type: stop
      hibernate: true
```
